data-collection/src/main/java/refactoringml/RefactoringAnalyzer.java

```diff
@@ -212,12 +212,18 @@ private Yes calculateCkMetrics(String refactoredClass, String refactorCommit, Ca
 		new CK().calculate(tempDir, ck -> {
 
 			String cleanedCkClassName = cleanClassName(ck.getClassName());
+			boolean isSubclass = ClassMetric.evaluateSubclass(ck.getClassName());
 
-			if(!cleanedCkClassName.equals(refactoredClass))
+			if(!isSubclass && !cleanedCkClassName.equals(refactoredClass)) {
```

Comment:
Why do we need this `!isSubclass` guard condition? I think it should not be part of this if.

the `calculate()` method from CK will invoke the callback to all classes it finds. So, if a class A contains two sub-classes, A1 and A2, then, this callback will be invoked three times, one for each class.

At this point, we, however, only care about one specific class, i.e., the refactored one, so we have to kinda ignore the others, and that's why there's this `if` that basically ignores the ones we do not care.

Does that make sense? Maybe we should add a comment here explaining this, as this is definitely not clear.

Reply:
Okay, I got this and changed it.